src/app/modules/test/components/test-questions/test-questions.component.ts

```diff
@@ -1,10 +1,274 @@
+      } else {
+        this.singleAnswerQuestionTotalCountTopicWise = 0;
+        this.multiAnswerQuestionTotalCountTopicWise = 0;
+        this.testService.questionCountSingleAnswer.next(QuestionCountInitial);
+        this.testService.questionCountMultiAnswer.next(QuestionCountInitial);
+      }
+    });
+    this.cdr.detectChanges();
+  }
+
+  ngOnChanges() {}
+
+  setQuestionCounts() {
+    this.testQuestionsCountData.map((res) => {
+      if (res.topicId == this.topicId) {
+        this.singleAnswerQuestionTotalCountTopicWise = res.singleAnswerCount;
+        this.multiAnswerQuestionTotalCountTopicWise = res.multiAnswerCount;
+        this.testService.questionCountSingleAnswer.next(res.singleAnswer);
+        this.testService.questionCountMultiAnswer.next(res.multiAnswer);
+      }
+    });
+  }
+
+  createForm() {
+    this.form = this.formBuilder.group({
+      topicId: ['', Validators.required],
+      numberOfQuestions: ['', Validators.required],
+      weightage: ['', Validators.required],
+      oneMarkQuestionSingleAnswer: [0, Validators.required],
+      twoMarkQuestionSingleAnswer: [0, Validators.required],
+      threeMarkQuestionSingleAnswer: [0, Validators.required],
+      fourMarkQuestionSingleAnswer: [0, Validators.required],
+      fiveMarkQuestionSingleAnswer: [0, Validators.required],
+      oneMarkQuestionMultiAnswer: [0, Validators.required],
+      twoMarkQuestionMultiAnswer: [0, Validators.required],
+      threeMarkQuestionMultiAnswer: [0, Validators.required],
+      fourMarkQuestionMultiAnswer: [0, Validators.required],
+      fiveMarkQuestionMultiAnswer: [0, Validators.required],
+    });
+  }
+
+  ngAfterViewInit() {
+    this.testService.questionCountSingleAnswer.next(QuestionCountInitial);
+    this.testService.questionCountMultiAnswer.next(QuestionCountInitial);
+    this.cdr.detectChanges();
+  }
+
+  handleEditQuestionsDialog() {
+    const dialogConfig = new MatDialogConfig();
+    dialogConfig.panelClass = ['primary-dialog'];
+    dialogConfig.autoFocus = false;
+    dialogConfig.width = '980px';
+    this.dialog.open(EditQuestionComponent, dialogConfig);
+  }
+
+  handleDeleteQuestionsDialog(topicId: number) {
+    const dialogConfig = new MatDialogConfig();
+    dialogConfig.panelClass = ['confirmation-dialog'];
+    dialogConfig.autoFocus = false;
+    this.dialog
+      .open(DeleteConfirmationDialogComponent, dialogConfig)
+      .afterClosed()
+      .subscribe((res) => {
+        if (res == true) {
+          this.deleteTopicWiseQuestions.emit(topicId);
+        }
+      });
+  }
+
+  handleDeleteAllQuestionsDialog() {
+    const dialogConfig = new MatDialogConfig();
+    dialogConfig.panelClass = ['confirmation-dialog'];
+    dialogConfig.autoFocus = false;
+    this.dialog
+      .open(DeleteConfirmationDialogComponent, dialogConfig)
+      .afterClosed()
+      .subscribe((res) => {
+        if (res == true) {
+          this.deleteAllQuestions.emit();
+        }
+      });
```

Comment:
you can make handleDeleteQuestionsDialog , handleDeleteAllQuestionsDialog common method